### PR TITLE
Quarantine Flaky e2e Tests

### LINKED ISSUE #35661: Isolate flakes from the rest of tests

### PR DESCRIPTION
**Pull request by iethree**
Closes https://github.com/metabase/metabase/issues/35661

### Description

- Quarantines Flaky e2e tests in their own non-blocking workflow
- Should all be assigned to be fixed and only live there temporarily

Top 10 flakes | Top 6 failures on master
--- | ---
Screen Shot 2023-11-14 at 12 14 32 PM | Screen Shot 2023-11-14 at 12 14 51 PM

### CODE REVIEW

File: .github/actions/build-e2e-matrix/action.yml [hunk 1]
```diff
@@ -50,6 +50,7 @@ runs:
             ["visualizations-tabular", {} ],
             ["oss-subset", { edition: 'oss', context: "special" } ],
             ["slow", { runner: beefierRunner, context: "special" } ],
+            ["flaky", { runner: beefierRunner, context: "special" } ],
```
Comment (nemanjaglumac) [inline .github/actions/build-e2e-matrix/action.yml:53]: Historically, running same tests on different runners yielded completely different results. I'm concerned that we'd maybe mask the underlying problem (or even potentially discover new ones specific to BuildJet runners).

IMO, we should solve the flakiness for the slowest and the least powerful runners, which are currently the default GitHub runners.

Comment (nemanjaglumac) [inline .github/actions/build-e2e-matrix/action.yml:53]: What about flakes that can run using only OSS uberjar?

Comment (iethree) [reply .github/actions/build-e2e-matrix/action.yml:53]: I just counted, and there are only 12 such tests in our entire test suite. I think it's best not to handle this.

**Review by perivamsi — APPROVED** [review]

**Review by nemanjaglumac — APPROVED** [review]
Let's ship this 🚀

**Pull request merged**